libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/util/AthenzUtils.java

```diff
@@ -145,11 +144,45 @@ public static String extractPrincipalDomainName(final String principalName) {
      * @return service name, null if it's not the expected full principal service name
      */
     public static String extractPrincipalServiceName(final String principalName) {
-        int idx = principalName.lastIndexOf('.');
+        int idx = principalName.lastIndexOf(AuthorityConsts.ATHENZ_PRINCIPAL_DELIMITER_CHAR);
         if (idx == -1 || idx == 0 || idx == principalName.length() - 1) {
             return null;
         } else {
             return principalName.substring(idx + 1);
         }
     }
+
+    /**
+     * Split principal to domain and service
+     * @param  name principal
+     * @return      [domain, service], null if principal in invalid format
+     */
+    public static String[] splitPrincipalName(String name) {
```

Comment:
I think I made a comment earlier about this. it looks like this method already does the to lowercase conversion thus the caller doesn't need to call it again.

Reply:
agreed. related reply: https://github.com/yahoo/athenz/pull/943#discussion_r418565694, https://github.com/yahoo/athenz/pull/943#discussion_r418566601